src/models/text.ts

```diff
@@ -147,6 +147,12 @@ class Sign {
 }
 
 class SignInterpretation {
+    private static nextId = 0;
+    public static get nextAvailableId() {
+        SignInterpretation.nextId --;
```

Comment:
I don't have the time to read through the code here, but it looks like you decrement the nextId every time someone reads nextAvailableId, was that intended.  I don't love a getter that has a mutation side effect, or am I simply reading the code wrong?  My apologies if so.

Reply:
It is intended, yes. This is not a regular property, and nobody binds to it - this is the property that provides the next frontend-only ID for new objects. If someone calls it too much, we'll just get holes - that's fine, we have about 2^31 negative numbers (maybe 2^63), and this is reset whenever you refresh the page.